src/meta/service/src/network.rs

```diff
@@ -133,18 +87,42 @@ impl Network {
         }
     }
 
-    pub fn with_backoff_policy(mut self, policy: BackOffPolicy) -> Self {
+    /// Set exponential back off policy for meta service
+    ///
+    /// - `ratio`: delay increase ratio of meta
+    ///
+    ///   should be not smaller than 1.0
+    ///
+    /// - `min_delay`: minimum back off duration, where the backoff duration vary starts from
+    ///
+    ///   random jitter between [0, min_back) will be set in each backoff, to minimize conflicts.
+    ///
+    /// - `max_delay`: maximum back off duration, if the backoff duration is larger than this, no backoff will be raised
+    ///
+    /// - `chances`: maximum back off times, chances off backoff
+    pub fn with_back_off_policy(
+        mut self,
+        ratio: f32,
+        min_delay: Duration,
+        max_delay: Duration,
+        chances: usize,
+    ) -> Self {
+        let policy = ExponentialBackoff::default()
+            .with_jitter()
+            .with_factor(ratio)
+            .with_min_delay(min_delay)
+            .with_max_delay(max_delay)
+            .with_max_times(chances);
+
         self.back_off_policy = policy;
         self
     }
 
-    pub(crate) fn backoff(
-        &self,
-    ) -> std::iter::Chain<ExponentialBackoff, std::vec::IntoIter<Duration>> {
+    pub(crate) fn back_off(&self) -> impl Iterator<Item = Duration> {
         // the last period of back off should be zero
-        // so the longest backoff will not be wasted
+        // so the longest back off will not be wasted
         let zero = vec![Duration::default()].into_iter();
-        self.back_off_policy.backoff().chain(zero)
+        self.back_off_policy.clone().chain(zero)
```

Comment:
I didn't get this. Why?

This function changes the semantic meaning of `backon::Backoff`:

https://docs.rs/backon/0.2.0/backon/trait.Backoff.html

> - `Some(Duration)` means caller need to sleep(Duration) and retry the same request
> - `None` means we have reaching the limits, caller needs to return current error instead.

We will have a `sleep(0)` for every backoff loop.

Reply:
Sorry for forgetting to `break` after a success request.

We are using ExponentialBackOff directly so things get different with just using a `backon::retry()` method or `backon::Retry` structure.

When the iteration just restarts from the longest back off, the next item in iterator will be `None`, then the `for` loop will be quited and the longest back off will be wasted.

To avoid this, to retry the same request after the longest back off, a zero duration sleep will be added. If the very last request failed, `sleep(0).await` cost little, the loop will reach the end and the error will be returned in time.

Reply:
> To avoid this, to retry the same request after the longest back off, a zero duration sleep will be added. If the very last request failed, `sleep(0).await` cost little, the loop will reach the end and the error will be returned in time.

I got it. Maybe we need to do something in `backon` to make using `Backoff` easier.

Reply:
Tracked at https://github.com/Xuanwo/backon/issues/23

Reply:
> Sorry for forgetting to `break` after a success request.

Oh, thank god that every consensus protocol tolerates duplicated messages. :D